### PR TITLE
bulkio: Produce better error message when parsing cron expression.

### LINKED ISSUE #53034: bulkio: Error message when incorrect cron syntax for schedules is extremely confusing / not helpful
When running something like `CREATE SCHEDULE ... RECURRING '* * haha'` it errors with `ERROR: missing field(s)` which is very confusing if it's in fact an invalid option / invalid cron syntax.

Error should help the user reach the best answer, either `Invalid schedule occurrence on 'RECURRING': Must be either valid cron or XYZ` something like that.

### PR DESCRIPTION
**Pull request by miretskiy**
Fixes #53034

Produce better error message when parsing cron expression.

Release Notes: None

### CODE REVIEW

File: pkg/ccl/backupccl/create_scheduled_backup.go [hunk 1]
```diff
@@ -145,7 +145,10 @@ func computeScheduleRecurrence(
 	}
 	expr, err := cronexpr.Parse(cron)
 	if err != nil {
-		return nil, err
+		return nil, errors.Newf(
```
Comment (mwang1026) [inline pkg/ccl/backupccl/create_scheduled_backup.go:148]: Will it be the same message for `RECURRING` and `FULL BACKUP`?

File: pkg/ccl/backupccl/create_scheduled_backup.go [hunk 2]
```diff
@@ -145,7 +145,10 @@ func computeScheduleRecurrence(
...
+			`error parsing schedule expression: %q.
+See https://en.wikipedia.org/wiki/Cron for more details.
```
Comment (mwang1026) [inline pkg/ccl/backupccl/create_scheduled_backup.go:150]: I don't think we need to link to the cron wikipedia page. I think people know what cron is. We could `must be valid cron syntax or valid option (see docs)` or something.

**Review by mwang1026 — CHANGES_REQUESTED** [review]
Minor nits.

File: pkg/ccl/backupccl/create_scheduled_backup.go (hunk 1 above)
Comment (miretskiy) [reply pkg/ccl/backupccl/create_scheduled_backup.go:148]: well, this one will be set when we try to pick recurrence.  I added another file to this pr to make sure the error is the same(ish)

Comment (pbardea) [reply pkg/ccl/backupccl/create_scheduled_backup.go:148]: Is the error that we get back here useful? If so, it might be worth wrapping?

Comment (pbardea) [reply pkg/ccl/backupccl/create_scheduled_backup.go:148]: By here, I mean the error we get back from `cronexpr.Parse`.

File: pkg/ccl/backupccl/create_scheduled_backup.go (hunk 2 above)
Comment (miretskiy) [reply pkg/ccl/backupccl/create_scheduled_backup.go]: Sure -- changed.

File: pkg/ccl/backupccl/create_scheduled_backup.go (hunk 1 above)
Comment (miretskiy) [reply pkg/ccl/backupccl/create_scheduled_backup.go:148]: No, it's not useful at all.

**Review by pbardea — APPROVED** [review]

**Review by mwang1026 — APPROVED** [review]
lgtm. thanks

**Pull request merged**